agent/app/agent_windows.go

```diff
@@ -76,12 +76,12 @@ func (agent *ecsAgent) initializeTaskENIDependencies(state dockerstate.TaskEngin
 		return err, false
 	}
 
-	// Query the VPC's primary IPv4 CIDR using IMDS
-	primaryIPv4VPCCIDR, err := agent.ec2MetadataClient.PrimaryIPV4VPCCIDR(agent.mac)
+	dnsServerList, err := agent.resourceFields.NetworkUtils.GetDNSServerAddressList(agent.mac)
 	if err != nil {
-		return fmt.Errorf("unable to get primary ipv4 cidr of the vpc: %v", err), false
+		// An error at this point is terminal as the tasks need the access to domain controllers.
+		return fmt.Errorf("unable to get dns server addresses of instance eni: %v", err), true
 	}
-	agent.cfg.PrimaryIPv4VPCCIDR = primaryIPv4VPCCIDR
+	agent.cfg.InstanceENIDNSServerList = dnsServerList
```

Comment:
Can we keep both `PrimaryIPv4VPCCIDR` and `dnsServerList` in config?

Reply:
Both are not required. Since we are using DNS server list from the instance, therefore, primary VPC CIDR becomes redundant.